pkg/plugin/loader/loader.go

```diff
@@ -53,98 +51,142 @@ func LoadAllPlugins(namespace string, searchPath []string, selections []plugin.S
 			continue
 		}
 
-		definitions, err := scanPlugins(dir)
+		files, err := findPlugins(dir)
 		if err != nil {
-			return ret, err
+			return []plugin.Interface{}, errors.Wrapf(err, "couldn't scan %v for plugins", dir)
 		}
+		pluginFiles = append(pluginFiles, files...)
+	}
 
-		defns = append(defns, definitions...)
+	pluginDefs := []*pluginDefinition{}
+	for _, file := range pluginFiles {
```

Comment:
nit: the variable naming in this block is a little confusing -- maybe `pluginFiles` could be `pluginDefinitionFiles`, then `pluginDefs` could be `pluginDefinitions` and `pluginIface` could be `plugin`?

Reply:
`plugin`'s a package name, is the problem there

Reply:
how's `s/pluginIface/loadedPlugin`?

Reply:
yeah, that sounds good to me